src/pkg/util/nvidiautils/nvidiautils.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package nvidiautils
+
+import (
+	"bufio"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// GetNvidiaBindPath returns a string array consisting of filepaths of nvidia
+// related files to be added to the BindPaths
+func GetNvidiaBindPath(abspath string) []string {
+	var strArray []string
+	var bindArray []string
+	var searchArray []string
+	var commentID = regexp.MustCompile(`#`)
+	var soID = regexp.MustCompile(".so")
+
+	// use nvidia-container-cli (if present)
+	command, err := exec.LookPath("nvidia-container-cli")
+	if err == nil {
+		cmd := exec.Command(command, "list", "--binaries", "--ipcs", "--libraries")
+		out, err := cmd.Output()
+		if err == nil {
+
+			for _, line := range strings.Split(string(out), "\n") {
+				val := soID.FindString(line) // this will disallow binaries (non .so files)
+				if val != "" {               // contains .so
+					if line != "" {
+						// extract the filename from the path
+						fileNames := strings.SplitAfter(line, "/")
+						fileName := fileNames[len(fileNames)-1]
+
+						testString := line + ":/.singularity.d/libs/" + fileName
+						bindArray = append(bindArray, testString)
+						strArray = append(strArray, fileName)
+					}
+				} else { // binary executable
+					bindArray = append(bindArray, line)
+					strArray = append(strArray, line)
+				}
+			}
+		}
+	}
+
+	cliEntries := strings.Join(strArray, " ") // save away for later comparison check (disallow duplicates)
+
+	// grab the entries in nvliblist.conf file
+	// use ldconfig to pattern match from ld.so.cache
+	newpath, err := filepath.Glob(abspath + "/nvliblist.conf")
```

Comment:
Oh so we are now using the _union_ of what we find in the conf file and what `nvidia-container-cli` yields?  I'm not sure I like that.  Kindof think we should default to `nvidia-container-cli` output and only fall back to conf file if it fails.

Reply:
nvidia-container-cli didn't provide everything though right? I thought there was some stuff missing.

Reply:
Well, the nvidia-container-cli function only returns the /dir/file.so.xxx.xx entries (where the xxx.xx is the driver version). I can't just strip off the .xxx.xx and use the so entries because some things (like nvidia-smi binary look for the file.so.1 entry). I could make it work if that is the way you think I should go...

Reply:
I think what you want to do is use _either_ the output of `nvidia-container-cli` _or_ the output of the `nvidia.conf` file as a set of things to search the `ld.so.cache` for (through `ldconfig` output).  

Does that make sense?  In other words if you get usable output from `nvidia-container-cli` skip parsing the `nvidia.conf` file and just proceed directly to pattern matching between `nvidia-container-cli` output and `ldconfig` output.

Reply:
If you want to just append `.1` to all of the `nvidia-container-cli` entries and search for them as well, I think it's reasonable?  

@jmstover yes there was a recent issue where a user was confused b/c `nvidia-container-cli` does not provide open GL libraries, but that behavior is correct and our behavior is wrong.  We should comment those libs out of the conf file imho because those libs are not strictly tied to the version of the driver as the other libs are.

Reply:
@GodloveD 
Maybe not on _recent_ drivers with libglvnd. But my `375.39` driver version, the GL libs *are* tied to the driver version.

i.e.  libGLX_nvidia.so.375.39

Reply:
@ctmadison I just had a brain wave.  I think you need to strip driver version from the `nvidia-container-cli` output and then use that as a _prefix_ when searching the output of `ldconfig`.  Tht way you will get any stray `.so.1` files (and other file appends) too.

Reply:
Yes, that should work. But, there is still the issue that @jmstover brought up about the GL libs that we have to agree on.